backend/app/repositories/user_repository.py

```
@@ -0,0 +1,17 @@
+        try:
+            result = await self.collection.insert_one(user_doc)
+            return result
+        except Exception as e:
+            raise Exception(f"Database insert failed: {e}")
+
+    async def find_user_by_key(self, user_key):
+        try:
+            user = await self.collection.find_one({"user_key": user_key})
+            return user
+        except Exception as e:
+            raise Exception(f"Database query failed: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)